renderer/src/common/components/theme/theme-provider.tsx

```diff
@@ -24,19 +24,16 @@ export function ThemeProvider({
     return isValidTheme(storedTheme) ? storedTheme : defaultTheme
   })
 
-  // Sync with Electron's native theme on mount
   useEffect(() => {
     const syncWithNativeTheme = async () => {
       if (window.electronAPI?.darkMode) {
         try {
-          const nativeThemeState = await window.electronAPI.darkMode.get()
-          const nativeThemeSource = nativeThemeState.themeSource
-
-          // Only sync if the stored theme doesn't match the native theme
           const storedTheme = localStorage.getItem(storageKey)
-          if (!isValidTheme(storedTheme) || storedTheme !== nativeThemeSource) {
-            setTheme(nativeThemeSource)
-            localStorage.setItem(storageKey, nativeThemeSource)
+
+          if (!isValidTheme(storedTheme)) {
+            const nativeThemeState = await window.electronAPI.darkMode.get()
+            setTheme(nativeThemeState.themeSource)
+            localStorage.setItem(storageKey, nativeThemeState.themeSource)
```

Comment:
Shouldn't be saved into file system directly? 🤔

Reply:
I think that is a great idea, we should have some local YAML config file system or sth like for this, and also for other features.